[IAC-1753] Update CIS compliance guide

ina-stoyanova opened the pull request:
Add (if missing) and review the below sections:
-> we can copy& paste directly from the LZ guide on the website, but be careful of links, code & any information that needs changing.

- [x] Pre-requisites @robmorgan 
- [x] The Gruntwork solution @robmorgan
- [x] Added: Create the root account @Ina
- [x] Replace with LZ instructions: Complete the manual steps @Ina
- [x] Create an IAM user in the root account @Ina
- [x] Lock down the root account IAM users @Ina
- [x] Configure the security baseline for the root account @Ina
- [x] Import existing resources from the root account @Ina
- [x] Apply the security baseline to the root account @Ina
- [x] Reset the root user password in each child account @Ina
- [x] Lock down the root user in the child accounts @Ina
- [x] Apply the security baseline to the logs account @Ina
- [x] Apply the security baseline to the security account @Ina
- [x] Apply the security baseline to the other child accounts @Ina
- [x] Configure AWS Security Hub in the root account @robmorgan 
- [x] Try authenticating as an IAM user to the child accounts @robmorgan 
- [x] Use IAM roles for EC2 instances (?) @robmorgan 
- [x] Maintaining compliance by following IAM best practices
- [x] Maintaining compliance by following Storage best practices
  - [x] S3 Buckets
  - [x] Encrypt EBS volumes [_Remove this section_] @robmorgan 
- [x] Maintaining compliance by following Logging best practices
  - [x] Cloudtrail [_Remove this section_] @robmorgan 
  - [x] AWS Config [_Remove this section_] @robmorgan 
  - [x] Enable key rotation for KMS keys
  - [x] Create VPC flow logs @robmorgan 
- [x] Maintaining compliance by following Monitoring best practices
  - [x] Subscribe to SNS topics
- [x] Maintaining compliance by following Networking best practices
  - [x] VPC Networks
- [x] Rewrite the Traceability matrix (_leave until last_) @infraredgirl

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+There you should be able to click on `Configure Security Challenge Questions` and add your questions.
+
+Enable MFA::
+Make sure to
+https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html#id_root-user_manage_mfa[enable MFA for your root user].
+Feel free to use a virtual or hardware MFA device—whichever is easier or required by your company—as either one
+dramatically improves the security of your root user. It is up to your discretion to decide which option yis suitable for your use case. CIS 1.3 bechmark recommends that any virtual or hardware device you use to fulfil MFA is not personal, but instead dedicated or company-owned.
```
this now also suggests that hardware or virtual MFA is desired.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
   region = var.aws_region
+
+  # Require a 3.x version of the AWS provider
+  version = "~> 3.40"
```
I've upped the version of the AWS provider here because that's what we seem to be doing the RefArches with - e.g. Nomisma `lock.hcl` file

robmorgan replied:
Makes sense.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+
+  # Only allow this Terraform version. Note that if you upgrade to a newer version, Terraform won't allow you to use an
+  # older version, so when you upgrade, you should upgrade everyone on your team and your CI servers all at once.
+  required_version = "= 0.14.8"
```
also upped the TF version here - this is what I've been testing with. Should it be something else? I assume once this repo is also upgraded to TF 0.15, we'll need to update this anyway

robmorgan replied:
Yep, leave it as 0.14.x for now and we can bump to 0.15 afterwards.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+  cloudtrail_force_destroy = var.force_destroy
+  config_force_destroy     = var.force_destroy
+
+  # Enable IAM Access Analyzer
+  iam_access_analyzer_type   = var.iam_access_analyzer_type
+  iam_access_analyzer_name   = var.iam_access_analyzer_name
+  enable_iam_access_analyzer = var.enable_iam_access_analyzer
```
This is taken from the `cis-repo/examples/ladningzone/account-baseline-root`.
I believe the IAM Access Analyzer should not be configurable:

```suggestion
  # Enable IAM Access Analyzer
  iam_access_analyzer_type   = var.iam_access_analyzer_type
  iam_access_analyzer_name   = var.iam_access_analyzer_name
```

robmorgan replied:
Yeah agreed, I imagine @marinalimeira probably hasn't updated her example yet.

ina-stoyanova replied:
I'll follow up and leave a comment on the `root` PR

ina-stoyanova replied:
this was already covered in the root PR.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+  description = "A feature flag to enable or disable this module."
+  type        = bool
+  default     = false
+// TODO -> this needs to be true, as IAM Access Analyzer needs to be enabled for CIS across all regions. Maybe we should even remove this variable from the example?
```
Either remove this variable, or set it at least to `true` by default. See CIS 1.3 requirement #1.21

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+# CONFIG OUTPUTS
+# ---------------------------------------------------------------------------------------------------------------------
+
+//TODO -> this needs fixing in the example - currently not referencing a real output
+//output "config_s3_bucket_name" {
+//  description = "The name of the S3 bucket used by AWS Config to store configuration items."
+//  value       = module.root_baseline.config_s3_bucket_name
+//}
+//
+//output "config_s3_bucket_arn" {
+//  description = "The ARN of the S3 bucket used by AWS Config to store configuration items."
+//  value       = module.root_baseline.config_s3_bucket_arn
+//}
```
These are currently not recognised & thus is invalid tf. This has been taken from `cis-repo/examples/landingzone/account-baseline-root`

robmorgan replied:
Ok, we are probably still waiting to update this example.

ina-stoyanova replied:
Yep, I thought so. I'll leave the related comments in this PR to check back on this.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+  description = "The ARN of the S3 bucket where cloudtrail logs are delivered."
+  value       = module.root_baseline.cloudtrail_s3_bucket_arn
+}
+//todo -> same as last
+//output "cloudtrail_kms_key_arn" {
+//  description = "The ARN of the KMS key used by the S3 bucket to encrypt cloudtrail logs."
+//  value       = module.root_baseline.cloudtrail_kms_key_arn
+//}
+//
+//output "cloudtrail_kms_key_alias_name" {
+//  description = "The alias of the KMS key used by the S3 bucket to encrypt cloudtrail logs."
+//  value       = module.root_baseline.cloudtrail_kms_key_alias_name
+//}
```
same here: 

These are currently not recognised & thus is invalid tf. This has been taken from cis-repo/examples/landingzone/account-baseline-root

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
-
-[[wrapper_code_terragrunt]]
-===== Using the compliance wrapper module with Terragrunt
+NOTE: This guide will use https://github.com/gruntwork-io/terragrunt[Terragrunt] and its associated file and folder
+structure to deploy Terraform modules. Please note that *Terragrunt is NOT required for using Terraform modules from
+the Gruntwork Infrastructure as Code Library.* Check out link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library[How to use the Gruntwork Infrastructure as Code Library]
+for instructions on alternative options, such as how to
+link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library#deploy_using_plain_terraform[deploying how to use plain terraform].
```
is this what we want for this guide too? None of our examples are currently with terragrunt.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+  # Prefix all resources with this name
+  name_prefix = "<COMPANY_NAME>-root"
+
+  # If you've already created an AWS Organization in your root account, you'll be able to import it later in this guide
```
```suggestion
  # If you've already created an AWS Organization in your root account, set this to false and you'll be able to import it later in this guide.
```

robmorgan commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,130 +857,683 @@ If you need to brush up on how the IaC Library works, read the
+=== Configure the security baseline for the root account
+
+Next, we'll configure a security baseline for the root account that is responsible for creating all the child accounts.
+It will also configure AWS Organizations, IAM Roles, IAM Users, IAM Groups, IAM Password Policies, Amazon GuardDuty, AWS CloudTrail, AWS Config and Security Hub.
```
Ok great!

robmorgan commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -951,128 +951,589 @@ dramatically improves the security of your IAM user. Note that using SMS (text m
 
-===== Using the wrapper module with Terraform
-If you're using Terraform without Terragrunt, use this section to create a module in your `infrastructure-modules` repository.
+We'll be using the `account-baseline-root` module from https://github.com/gruntwork-io/terraform-aws-service-catalog[terraform-aws-service-catalog].
```
I think the money shot here is actually using the `account-baseline-root` module from the CIS service catalog instead. I'll go ahead and change it!

robmorgan replied:
Actually don't worry, I can see you've changed it in a subsequent commit.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1283,28 +3275,27 @@ To limit the number of flow logs, you may want to use the
 link:https://github.com/gruntwork-io/cloud-nuke[`cloud-nuke defaults-aws`] command. It will remove the default VPC from
 all regions in an account, saving you the hassle of creating flow logs in each default VPC.
 
-[[configure_monitoring]]
```
I feel like in this section we should also include the IAM Access Analyzer & the Security Hub too. My thinking is that they're also monitoring tools for any security issues. Curious to get more thoughts on this :)

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
- +
- +
+To further simplify and expedite compliance, the Gruntwork CIS Service Catalog uses both standalone compliance
+modules as well as "wrappers" around the core, unopinionated modules in the Infrastructure as Code Library. The
+standalone compliance modules are designed with the compliance requirements built-in, whereas the wrappers
+call the core modules with configuration values that are compliant with the AWS Foundations Benchmark. You can use both
+types of modules by creating a module of your own (in the case of the wrapper modules, this can be considered a second
```
NIT: I understand what this says, but I had to read it a few times to grasp it. Might be my English 😛 but it would make it easier if we said something like this:
```suggestion
modules as well as "wrappers" around the core, unopinionated modules in the Infrastructure as Code Library. The
standalone compliance modules are designed with the compliance requirements built-in, whereas the wrappers
call the core IaC modules and pass down configuration values that enforce the AWS Foundations Compliance Benchmark. You can use both
types of modules by creating a module of your own (in the case of the wrapper modules, this can be considered a second
```

ina-stoyanova replied:
or something similar - just to make it explicit which modules are which.

ina-stoyanova replied:
happy to leave it as it is now

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+ +
+
+.Types of CIS module relationships to avoid repetitive code and minimize the amount of extra work needed to achieve compliance.
+image::/assets/img/guides/cis-compliance/cis-module-relationships.png[]
```
nice diagram @robmorgan!

rhoboat replied:
I'm not sure if users will understand that the `infrastructure-live` repo is optional because this walkthrough is written entirely using that repo. Hmm..

rhoboat replied:
I suppose the section below, which talks about how to Create your own service catalog, is sufficient to clarify this point.

robmorgan replied:
@rhoboat ok!

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+your credentials) or for the
+https://docs.aws.amazon.com/general/latest/gr/aws_tasks-that-require-root.html[small number of tasks that require root user credentials].
+
+_This fulfils requirement #1.7 from the CIS 1.3.0 benchmark._
```
```suggestion
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+https://www.theverge.com/2017/9/18/16328172/sms-two-factor-authentication-hack-password-bitcoin[vulnerabilities with the cellular system],
+so using a virtual or hardware MFA device is preferable; that said, MFA with SMS is still better than no MFA at all.
+
+wrapper module]. Complete this step before creating any IAM users!
```
```suggestion
Complete this step before creating any IAM users!
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+----
+
+So the `<ADDRESS>` you want is `module.root_baseline.module.iam_users.aws_iam_user.user["alice"]`. Now, normally, you'd
+run `import` right away, but due two Terraform bugs, https://github.com/hashicorp/terraform/issues/13018[#13018] and
+https://github.com/hashicorp/terraform/issues/26211[#26211], `import` doesn't work on certain types of modules—namely,
```
NIT: It seems like these issues have a fix now. The first one has a fix shown in another issue: https://github.com/hashicorp/terraform/issues/27934 and here https://github.com/hashicorp/terraform/issues/26211#issuecomment-805270266

ina-stoyanova replied:
We might need to test the fix and update the guide.

ina-stoyanova replied:
raise a new issue for this, so resolving comment.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+  # The AWS region in which all resources will be created
+  region = var.aws_region
+
+  # Require a 2.x version of the AWS provider
+  version = "~> 2.6"
```
```suggestion
  # Require a 3.x version of the AWS provider
  version = "~> 3.40"
```

ina-stoyanova replied:
to be consistent with the rest of the AWS provider versions

ina-stoyanova reviewed (commented):
Overall, I think the documentation makes sense the way it is! The outstanding things according to [this](https://gruntwork-io.slack.com/archives/C01LR3Q5LJH/p1622635859011700) plan are:
- traceability matrix that Ana's looking at 
- security hub part that Rob has got in progress

I was thinking to carve out some time and go through the docs from scratch & follow the instructions. Do you think there's much value in doing that?

infraredgirl commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2409 @@ Terraform::
+https://www.theverge.com/2017/9/18/16328172/sms-two-factor-authentication-hack-password-bitcoin[vulnerabilities with the cellular system],
+so using a virtual or hardware MFA device is preferable; that said, MFA with SMS is still better than no MFA at all.
+
+Complete this step before creating any IAM users!
```
This sentence is confusing - which step is it referring to? Could we clarify this a bit?

ina-stoyanova replied:
you're right - it's a bit out of nowhere.... not sure what it was referring to, but now I'd vote let's remove this line for now.

robmorgan replied:
I'm removing for now!

infraredgirl commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2409 @@ Terraform::
+  # IAM users to create in this account
+  users = var.users
+
+  # These are variables you only need to set at test time so that everything can be deleted cleanly.
+  # You will likely NOT need to set this in any real environments.
+  force_destroy_users      = var.force_destroy
+  cloudtrail_force_destroy = var.force_destroy
+  config_force_destroy     = var.force_destroy
```
Should we be mentioning them here then, considering this is a production guide?

ina-stoyanova replied:
I think you might be right and we don't want these here!

infraredgirl commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2409 @@ Terraform::
+
+The example code above does the following:
+
+. **Create 6 child AWS accounts**. These are the accounts described in the <<child_accounts>> sections.
```
The `<<child_accounts>>` link does not resolve.

ina-stoyanova replied:
Same here: This refers to the Standard Service catalog actually. I wonder if we need to have such a section in our CIS guide too.. https://gruntwork.io/guides/foundations/how-to-configure-production-grade-aws-account-structure/#child_accounts

infraredgirl commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2409 @@ Terraform::
 
-If you're using Terragrunt, you can use the password policy Terraform module in the CIS compliance repository from a Terragrunt configuration directly, with no need to create a Terraform module of your own! In this case, `terraform-aws-cis-service-catalog` is the canonical Terraform module. To do so, configure a `terragrunt.hcl` to invoke the module. First, create a directory for the password pol ...
+Now that your security account is fully configured, you need to apply the security baseline to the remaining child
+accounts (e.g., dev, stage, prod, shared-services) as detailed in the <<child_accounts>> section. Feel free to adjust
```
The `<<child_accounts>>` anchor does not resolve.

ina-stoyanova replied:
This refers to the Standard Service catalog actually. I wonder if we need to have such a section in our CIS guide too.. https://gruntwork.io/guides/foundations/how-to-configure-production-grade-aws-account-structure/#child_accounts

robmorgan replied:
I've reworked the "child accounts" sections.

infraredgirl commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -763,372 +763,2410 @@ Terraform::
+source_profile=root-iam-user
 ----
 
-[[iam_access_analyzer_deployment]]
```
This section has been removed but is still being linked to from other parts of the document. We should reconcile that.

ina-stoyanova replied:
Hm, looking at that - it's just that we're no longer deploying that on its own. We're now using the baselines to deploy that. So maybe just remove the refs to this link & make sure the sentences make sense.

robmorgan replied:
@ina-stoyanova @infraredgirl I spoke to Ana and we agreed to resolve this.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1752,6 +1751,7 @@ Once you're able to access the root user of each child account, you should follo
 for each of those child accounts—including enabling MFA and deleting the root user's access keys—and (almost) never use
 those root users again.
 
+[[apply_account_baseline_logs]]
```
not just here, but nicely done on the consistent formatting 👍 😄

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
 1.2,<<security_questions>>,Complete the security contact information on the AWS account page
 1.3,<<security_questions>>,Answer the security questions on the AWS account page
-1.4,<<next_steps>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that no access key exists for the root user
+1.4,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that no access key exists for the root user
```
**_Context:_** Technically, I believe Security Hub isn't part of the CIS official requirements. It is, however, a useful tool to monitor that for example this is a fulfilled rule. It checks currently against CIS 1.2 benchmark, not even 1.3.

infraredgirl replied:
True. If there's a better section to link to here, let me know.

ina-stoyanova replied:
I think by adding the section for Security hub, this makes more sense now

robmorgan replied:
Yeah, it looks like it's required in CIS 1.4.0, so we'll probably be adding it in the future anyway.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
 1.2,<<security_questions>>,Complete the security contact information on the AWS account page
 1.3,<<security_questions>>,Answer the security questions on the AWS account page
-1.4,<<next_steps>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that no access key exists for the root user
+1.4,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that no access key exists for the root user
```
```suggestion
1.4,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure you are notified if the root user has access keys set.
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
 1.6,<<root_mfa>>,Use a Yubikey (or other hardware MFA) for the root user
 1.7,<<manual_steps>>,Take manual steps to complete this recommendation
-1.8-9,<<create_password_policy>>,Use the IAM password policy module
+1.8-9,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users
```
```suggestion
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
-1.16,<<iam_user_authentication>>,Use the Gruntwork modules to create best-practices groups and roles
-1.17,<<iam_user_authentication>>,Use the `iam-groups` module to create a support group
+1.11,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users
+1.12,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
```
```suggestion
1.12,<<apply_account_baseline_security>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
```

ina-stoyanova replied:
I believe the Security Hub monitors this, but we have an AWS Config rule for this that should get kicked in

infraredgirl replied:
This is a bit tricky, since those are links pointing to the specific sections in the guide, so `apply_account_baseline_all` won't work. I guess I could put multiple links pointing to each account baseline module

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
-1.17,<<iam_user_authentication>>,Use the `iam-groups` module to create a support group
+1.11,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users
+1.12,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
+1.13,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no extra access keys
```
Same as above:

```suggestion
1.13,<<apply_account_baseline>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no extra access keys
```

ina-stoyanova replied:
Same as above.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+1.11,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users
+1.12,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
+1.13,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no extra access keys
+1.14,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused access keys
```
Same as above.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+1.12,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused credentials
+1.13,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no extra access keys
+1.14,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused access keys
+1.15,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users and groups
```
👍

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+1.13,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no extra access keys
+1.14,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused access keys
+1.15,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users and groups
+1.16,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create best-practices groups and roles
```
👍

ina-stoyanova replied:
We could be more explicit here and say something like "to ensure no full-access policies are attached to any groups or users"

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+1.14,<<configure_security_hub>>,Use the Gruntwork Security Hub module to enable AWS Security Hub to ensure that there are no unused access keys
+1.15,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create users and groups
+1.16,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create best-practices groups and roles
+1.17,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to create a support group
```
👍

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+2.1.1-2.1.2,<<s3_buckets_deployment>>,Use the `private-s3-bucket` module
+2.2.1,<<configure_ebs_encryption>>,Use Gruntwork modules to configure AWS EBS encryption
+3.1-3.4,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
+3.5,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to ensure AWS Config is enabled
```
```suggestion
3.5,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to ensure AWS Config is enabled in all regions
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
+2.2.1,<<configure_ebs_encryption>>,Use Gruntwork modules to configure AWS EBS encryption
+3.1-3.4,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
+3.5,<<apply_account_baseline_security>>,Use the `account-baseline-security` module to ensure AWS Config is enabled
+3.6-3.7,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
```
```suggestion
3.6<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail S3 bucket has enabled access logging```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
-5.2,<<configure_networking>>,Use the Gruntwork CIS-compliant `vpc` service for a secure network configuration
-5.3,<<configure_networking>>,Use the `cloud-nuke` tool to remove all default security groups
-5.4,<<configure_networking>>,Use the Gruntwork CIS-compliant `vpc` service to configure least-privilege routing by default
+3.10-3.11,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
```
```suggestion
3.10-3.11,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Object-level logging is enabled for S3 buckets for read and write events
```

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
-5.3,<<configure_networking>>,Use the `cloud-nuke` tool to remove all default security groups
-5.4,<<configure_networking>>,Use the Gruntwork CIS-compliant `vpc` service to configure least-privilege routing by default
+3.10-3.11,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
+4.1-4.15,<<maintain_compliance_monitoring>>,The CloudWatch Logs metrics filters wrapper module will satisfy each recommendation
```
👍

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -3357,35 +3357,35 @@ sections above.
-5.4,<<configure_networking>>,Use the Gruntwork CIS-compliant `vpc` service to configure least-privilege routing by default
+3.10-3.11,<<apply_account_baseline_logs>>,Use the `account-baseline-*` modules to ensure Cloudtrail is enabled and configured in all regions
+4.1-4.15,<<maintain_compliance_monitoring>>,The CloudWatch Logs metrics filters wrapper module will satisfy each recommendation
+5.1,<<maintain_compliance_networking>>,Use the Gruntwork CIS-compliant `vpc` service to ensure there is no public remote access
+5.2,<<maintain_compliance_networking>>,Use the Gruntwork CIS-compliant `vpc` service for a secure network configuration
+5.3,<<maintain_compliance_networking>>,Use the `cloud-nuke` tool to remove all default security groups
+5.4,<<maintain_compliance_networking>>,Use the Gruntwork CIS-compliant `vpc` service to configure least-privilege routing by default
```
nice

ina-stoyanova reviewed (commented):
I've reviewed the matrix - but would be useful to chat through some of the comments I've left just to make sure my understanding was correct.

infraredgirl commented:
Thanks, @ina-stoyanova - I addressed your comments!

ina-stoyanova commented:
Ok, I've had another look at the last few comments - think I've sorted them out:
- a few broken links for individual modules deployments - I removed those, and replaced with a generic message to refer to the account baseilnes deployment steps
- a few comments for us - to cleanup sections:
   - the Security hub section makes sense to me. Is there anything else to add to it? 
   - to the question above - I think we should actually mention it's deployed automatically by the account baselines modules

marinalimeira reviewed (commented):
NIT (line 134 and 135): change URLs to https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-user-guide.html and https://docs.aws.amazon.com/config/latest/developerguide/WhatIsConfig.html.

marinalimeira commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,473 +856,2421 @@ If you need to brush up on how the IaC Library works, read the
+the central place where you manage billing. You create this initial account manually, via a web browser:
+
+. Go to https://aws.amazon.com.
+. Click Create an AWS Account.
```
This button doesn't appear for me. It appears only after "Sign In to the Console"

robmorgan replied:
@marinalimeira This is what I see in the top-right. Are we okay to resolve this one?

<img width="1515" alt="Screen Shot 2021-06-09 at 10 50 03 am" src="https://user-images.githubusercontent.com/178939/121285185-843a3900-c910-11eb-8d14-135e5fa62aba.png">

marinalimeira commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,473 +856,2421 @@ If you need to brush up on how the IaC Library works, read the
+huge problem for your company (especially if that IAM user had admin permissions), so it's still critical to lock down
+IAM user accounts as much as possible:
+
+Use a secrets manager::
```
This section seems duplicated, just a NIT for we to reduce the amount of text in this article

robmorgan replied:
@marinalimeira Overall I agree with you, but as it's only 3 steps I'd rather leave them "as-is" to avoid confusion. Otherwise for "Lock down the root account IAM users", we'd have to say follow the recommendations in "Lock down the root user", but ignore all of them except for:

* Use a secrets manager
* Use a strong, generated password
* Enable MFA

So it ends up getting slightly verbose anyway.

robmorgan replied:
@ina-stoyanova @marinalimeira resolving by leaving it in for now.

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -458,7 +458,7 @@ As of version 1.3.0, the CIS recommendations stipulate that the AWS IAM Access A
 
 To achieve this compliance requirement, enable the IAM Access Analyzer service for every AWS region you have enabled in
-every one of your AWS accounts. Alternatively, you could make use of the `iam-access-analyzer-multi-region` module
+every one of your AWS accounts. Alternatively, you could make use of the link:https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/iam-access-analyzer-multi-region[`iam-access-analyzer-multi-region`] module
```
nice 👍

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -557,10 +557,10 @@ resource "aws_ebs_encryption_by_default" "ebs_encryption" {
-* link:https://aws.amazon.com/config/[AWS Config]
-* KMS key rotation
-* VPC Flow logs
+* link:https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-user-guide.html[AWS CloudTrail]
+* link:https://docs.aws.amazon.com/config/latest/developerguide/WhatIsConfig.html[AWS Config]
+* link:https://docs.aws.amazon.com/kms/latest/developerguide/rotate-keys.html[KMS key rotation]
+* link:https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs.html[VPC Flow Logs]
```
also a nice touch!

robmorgan replied:
Yeah good idea, that makes them consistent!

ina-stoyanova commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -2217,11 +2218,6 @@ inputs = {
-                                               account_id = "<LOGS_ACCOUNT_ID>"
-                                               email      = "root-accounts+logs@acme.com"
-                                             }
-                                        }
```
I'm wondering why this is removed? Is it no longer necessary?

marinalimeira replied:
Because this is set in the root account. This account only needs to add the root account id.

robmorgan replied:
@ina-stoyanova @marinalimeira let me know if we can resolve this one?

ina-stoyanova reviewed (commented):
Looks good @marinalimeira

robmorgan commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -459,8 +459,8 @@ AWS Account or Organization.
+deployment walkthrough to see how this module is used from the Landing Zone account baselines and is deployed in each of your AWS accounts.
 
 Once enabled, it will scan only within the boundaries of the AWS Account or Organization it has access to. Only specific
 resources are analyzed and included in the results - e.g. S3 buckets, SQS, etc. (For the full list of resources supported,
```
@marinalimeira I'm going to remove the link. As you mentioned, it's broken and the only other similar resources I could find are on non-Amazon-controlled websites.

ina-stoyanova commented:
This is looking good! 👏  Besides the diagram update, what else is needed for us to publish this?

ina-stoyanova commented:
Ok @marinalimeira when you have a chance, could you have a look and let me know if this is what you meant ([I mean this commit](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/426/commits/bbd42831233f7fa536147d4bd7c8ce656f0e16f1)) and we can go from there?

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -737,48 +737,55 @@ Infrastructure as Code modules from Gruntwork.
+==== How to configure a production-grade AWS account structure
+Review the link:https://gruntwork.io/guides/foundations/how-to-configure-production-grade-aws-account-structure/[production-grade AWS account structure guide] to familiarize yourself with many of the concepts that this walkthrough depends on.
+
+==== Terraform
+This guide uses https://www.terraform.io/[Terraform] to define and manage all the infrastructure as code. If
+you're not familiar with Terraform, check out
+https://blog.gruntwork.io/a-comprehensive-guide-to-terraform-b3d32832baca[A Comprehensive Guide to Terraform],
+https://training.gruntwork.io/p/terraform[A Crash Course on Terraform], and
+link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library[How to Use the Gruntwork
+Infrastructure as Code Library].
+
+==== Keybase (optional)
+As part of this guide, you will create IAM users, including, optionally, credentials for those IAM users. If you
+choose to create credentials, those credentials will be encrypted with a PGP key. You could provide the PGP keys
+manually, but a more manageable option may be to have your team members to sign up for https://keybase.io[Keybase],
+create PGP keys for themselves, and then you can provide their Keybase usernames, and the PGP keys will be retrieved
+automatically.
```
Is it necessarily to include Terragrunt here, as well as any new repositories they need to create, or do you think it's sufficient that we link them to the LZ guide, which will mention those things?

I notice there's terragrunt commands mentioned here, so maybe at least mentioning terragrunt would be a good idea?

rhoboat replied:
Ah, I see that the `infrastructure-live` repo is "optional" in the diagram below.. will comment there...

robmorgan replied:
hi @rhoboat, we considered the following options:

1. We include a minor mention of Terragrunt in a paragraph.
2. We basically copy everything that Rho wrote and reword the necessary bits.
3. We do nothing and update the guide later to link to Rho's section once it's published.
4. We do nothing.

In the end, we decided to include a small paragraph (stolen from your work) on Terragrunt. In the future, we could simply de-duplicate work and link to anything you may publish in the LZ guide but I was hesitate to introduce a dependency for the moment.

robmorgan commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
-  aws_account_id = "<LOGS_ACCOUNT_ID>"
+  # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = "us-east-1"
+  aws_account_id = local.accounts["logs"]
```
Nice 👍

rhoboat replied:
NIT: I believe this could simply be `local.accounts.logs`, as well.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
 [source,hcl]
 ----
+locals {
+  accounts = {
+    logs     = "409800740445"
+    security = "673123100581"
+    shared   = "384759303421"
+    dev      = "293847503945"
+    stage    = "384924092834"
+    prod     = "784260063686"
+    root     = "216044045972"
+  }
```
So, in the LZ guide, I've moved these into the root `accounts.json` that gets included in the root `common.hcl` as `accounts = jsondecode(file("accounts.json"))`. Then every `terragrunt.hcl` that needs it does

```
locals {
  # A local for more convenient access to the accounts map.
  accounts = local.common_vars.locals.accounts
  ...
}
```

It's a nit, but wonder if it's something you want to follow here.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
+  # All of the other accounts send logs to this account.
+  config_linked_accounts = [
+  for name, id in local.accounts :
+  id if name != "logs"
```
NIT: In my view, this is a line under the for loop, so it seems like it should be further indented. (We need to update the ref arch generation to reflect this too.)

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
+  cloudtrail_allow_kms_describe_key_to_external_aws_accounts = true
+  cloudtrail_external_aws_account_ids_with_write_access = [
+  for name, id in local.accounts :
+  id if name != "logs"
```
NIT: In my view, this is a line under the for loop, so it seems like it should be further indented. (We need to update the ref arch generation to reflect this too.)

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1599,49 +1723,109 @@ Set the variables for the `account-baseline-security` module in this environment
+input = {
+ # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = "us-east-1"
+  aws_account_id = local.accounts["security"]
```
NIT: This could be `local.accounts.security`.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
+  aws_region = "us-east-1"
+
+  accounts = {
+    logs     = "409800740445"
+    security = "673123100581"
+    shared   = "384759303421"
+    dev      = "293847503945"
+    stage    = "384924092834"
+    prod     = "784260063686"
+    root     = "216044045972"
+  }
```
NIT: See comment in `logs` section about DRYing this up.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
-  aws_account_id = "<STAGE_ACCOUNT_ID>"
+  # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = local.aws_region
+  aws_account_id = local.accounts["stage"]
```
NIT: This could be `local.accounts.stage`.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -992,16 +992,26 @@ Set the variables for the `account-baseline-root` module in this environment in
 [source,hcl]
 ----
+locals {
+  accounts = {
+    root = "216044045972"
+  }
```
NIT: Perhaps this ID could be stored as a `root` key in the `accounts.json` that I mention in the comment under the `logs` section below.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -992,16 +992,26 @@ Set the variables for the `account-baseline-root` module in this environment in
   # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
-  aws_region     = "us-east-2"
-  aws_account_id = "<ROOT_ACCOUNT_ID>"
+  aws_region     = "us-east-1"
```
NIT: This could be `local.aws_region`, I think.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
-  aws_region     = "us-east-2"
-  aws_account_id = "<LOGS_ACCOUNT_ID>"
+  # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = "us-east-1"
```
NIT: This could be `local.aws_region`, I think.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1599,49 +1723,109 @@ Set the variables for the `account-baseline-security` module in this environment
+}
+input = {
+ # Fill in your region you want to use (only used for API calls) and the ID of your root AWS account
+  aws_region     = "us-east-1"
```
NIT: This could be `local.aws_region`, I think.

marinalimeira replied:
This file is outdated, I updated this earlier this morning :D

marinalimeira replied:
So this is ✅

robmorgan replied:
I'll resolve this one.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1599,49 +1723,109 @@ Set the variables for the `account-baseline-security` module in this environment
+    root     = "216044045972"
+  }
+
+  # Both buckets are created in the logs account by account-baseline-root
+  config_s3_bucket_name     = "acme-config-bucket-logs"
+  cloudtrail_s3_bucket_name = "acme-cloudtrail-logs"
+
+  # The Cloudtrail KMS Key is deployed at the logs account but it's value is an output from the root account.
+  cloudtrail_kms_key_arn = "arn:aws:kms:us-east-1:216044045972:alias/cloudtrail-acme"
```
NIT: Is it worth replacing these hardcoded values with placeholder values, with a comment above them telling them how to get them? Like:

```
  # Use the S3 bucket and KMS key that were already created in the logs account by account-baseline-root
  cloudtrail_s3_bucket_name = "<CLOUDTRAIL_BUCKET_NAME>"
  cloudtrail_kms_key_arn    = "<CLOUDTRAIL_KMS_KEY_ARN>"
```

rhoboat replied:
I think what you have currently works fine. In either case, it's up to the customer whether they want to store the values in locals, and if you have them set it there, then it makes no difference that you reference the local here.

rhoboat replied:
I made this modification in #442, so we store the bucket names and arns in `common.hcl`. Updating the LZ guide, too.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1426,58 +1459,149 @@ Set the variables for the `account-baseline-app` module in this environment in t
+    root     = "216044045972"
+  }
+
+  # Both buckets are created in the logs account by account-baseline-root
+  config_s3_bucket_name     = "acme-config-bucket-logs"
+  cloudtrail_s3_bucket_name = "acme-cloudtrail-logs"
+
+  # The Cloudtrail KMS Key is deployed at the logs account but it's value is an output from the root account.
+  cloudtrail_kms_key_arn = "arn:aws:kms:us-east-1:409800740445:alias/cloudtrail-acme"
```
NIT: Same nit as below on replacing hardcoded values with placeholders.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
+    root     = "216044045972"
+  }
+
+  # Both buckets are created in the logs account by account-baseline-root
+  config_s3_bucket_name     = "acme-config-bucket-logs"
+  cloudtrail_s3_bucket_name = "acme-cloudtrail-logs"
+
+  # The Cloudtrail KMS Key is deployed at the logs account but it's value is an output from the root account.
+  cloudtrail_kms_key_arn = "arn:aws:kms:us-east-1:409800740445:alias/cloudtrail-acme"
```
NIT: Same nit about hardcoded values.

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
 
   # Prefix all resources with this name
-  name_prefix = "<COMPANY_NAME>-stage"
+  name_prefix = "stage-logs"
```
Hm, this one I'm not sure about. Setting the `name_prefix` for the environment + account is not what we're currently doing in the ref arch deploys. See the [LZ guide](https://gruntwork.io/guides/foundations/how-to-configure-production-grade-aws-account-structure/#prepare-your-infrastructure-live-repository) for how we set `name_prefix` (in `common.hcl`, usually as the company name, and then in the root `terragrunt.hcl`, so that no downstream `terragrunt.hcl`s need to set it). It makes me wonder if our ref arch is setting the `name_prefix` a little too generic? Pinging @bwhaley for another opinion.

bwhaley replied:
It does need to be something unique as it will ultimately be used to create the S3 bucket, which is a global namespace among all AWS users. Identifying a good rule of thumb is tricky because it must be both (a) short enough to not run in to length constraints, and (b) globally unique. `<COMPANY_NAME>` is usually unique enough, but runs the risk of being too long. Many customers have chosen a prefix that is an acronym or shortened identifiable version of the company name, like `phc` or `mf`. 

I think we ought to share this context in the comment on the preceding line, and go with something like:

```
name_prefix = "<SOME UNIQUE IDENTIFIER>-logs"
```

rhoboat replied:
Thanks Ben. Seems like we need to update the LZ guide and the Ref Arch code, too. Looking at the `terraform-aws-architecture-catalog`, we set `name_prefix` is only [one place](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/common.hcl#L13-L14), which gets passed around from the [root terragrunt](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/terragrunt.hcl#L25) (and actually it's [duplicated here](https://github.com/gruntwork-io/terraform-aws-architecture-catalog/blob/b1b7049d9d510d12d0c79ab70b0d0acf055a526c/templates/infrastructure-live/_root/terragrunt.hcl#L73), which we can simplify). That means it's not unique for the S3 buckets, right?

rhoboat replied:
I feel like I must be missing something. It can't be that we use the same `name_prefix` everywhere. Maybe it does get overridden in each account's terragrunt, but I'm not seeing it.

rhoboat replied:
@bwhaley Thoughts on this?

rhoboat commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -1810,37 +1993,159 @@ Set the variables for the `account-baseline-app` module in this environment in t
+  }
+  kms_grants = {
+    ami_encryption_key = {
+      kms_cmk_arn       = "arn:aws:kms:us-east-1:${local.accounts.shared}:alias/ami-encryption"
```
NIT: I might remove the reference to `us-east-1` here, using `local.aws_region` or some such, instead.
Or maybe this whole arn should be a placeholder text.

rhoboat reviewed (commented):
I just left a few comments after looking more in depth at the terragrunt code.

brikis98 commented on _posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc:
```diff
@@ -842,293 +846,1449 @@ If you need to brush up on how the IaC Library works, read the
+    "arn:aws:iam::${id}:root" if name != "security"
+  ]
+
+  # A list of account root ARNs that should be able to assume the auto deploy role.
+  allow_auto_deploy_from_other_account_arns = [
+    # External CI/CD systems may use an IAM user in the security account to perform deployments.
+    "arn:aws:iam::${local.accounts.security}:root",
+
+    # The shared account contains automation and infrastructure tools, such as CI/CD systems.
+    "arn:aws:iam::${local.accounts.shared}:root",
+  ]
+  auto_deploy_permissions = [
+    "iam:GetRole",
+    "iam:GetRolePolicy",
+  ]
```
Again, what auto deploy do we do here? Also, you're allowing access from the security account... But this _is_ the security account, so this doesn't quite make sense.

brikis98 reviewed (approved):
OK, just did a review pass. I found a bunch of NITs, which you can safely ignore for now, and perhaps implement in follow-up PRs. I also found a few places that look like legit bugs. These should probably be fixed ASAP. That said, I'll still mark this with a "ship it," as this PR has been open for ages, is still an improvement over the current guide, and even the "ASAP" fixes can still be done in follow-up PRs.


One more NIT: There are a very large number of sub-sections below deployment walkthrough:

<img width="270" alt="Screen Shot 2021-06-16 at 3 46 21 PM" src="https://user-images.githubusercontent.com/711908/122241100-326a5380-ceba-11eb-8b43-012f98bfa493.png">

That might scare users off... Is it worth grouping some of those into slightly larger sections?

ina-stoyanova commented:
Thanks @brikis98 @bwhaley & @rhoboat for the thorough reviews! 

I think here we should follow your suggestion:

ina-stoyanova commented:
Oops - before posting the last comment I pressed the wrong button and closed the PR by mistake! Apologies
